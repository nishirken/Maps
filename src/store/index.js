export testInitialState from './test-initial-state';
export testServerInitialState from './test-server-initial-state';
export store from './create-store';
export testStore from './test-store';
