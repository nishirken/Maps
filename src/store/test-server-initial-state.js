import { fromJS } from 'immutable';

export default fromJS([
    {
        index: 1,
        coords: {
            lng: 100,
            lat: 200,
            x: 20,
            y: 21,
        },
        name: 'Marker 1',
        objects: [
            {
                index: 0,
                name: 'Object 1',
            },
            {
                index: 1,
                name: 'Object 2',
            },
        ],
    },
    {
        index: 2,
        coords: {
            lng: 200,
            lat: 300,
            x: 10,
            y: 11,
        },
        name: 'Marker 2',
        objects: [
            {
                index: 0,
                name: 'Object 1',
            },
            {
                index: 1,
                name: 'Object 2',
            },
        ],
    },
]);
