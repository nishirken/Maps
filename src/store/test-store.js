import configureStore from 'redux-mock-store';
import initialState from './test-initial-state';

const mockStore = configureStore();
const store = mockStore(initialState.toObject());

export default store;
