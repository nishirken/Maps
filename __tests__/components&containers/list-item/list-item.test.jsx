import store from 'Store';
import { setCurrentMarker } from 'Actions';
import { ListItem } from 'Components';

describe('Marks list item', () => {
    const TestMarksLIstItemShallow = shallow(
        <ListItem
            setCurrentMarker={setCurrentMarker}
            store={store}
        />
    );

    it('Rendered', () => {
        expect(TestMarksLIstItemShallow).toMatchSnapshot();
    });
});
