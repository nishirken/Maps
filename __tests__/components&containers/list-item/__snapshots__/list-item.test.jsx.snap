exports[`Marks list item Rendered 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div
    current={undefined}
    onClick={[Function]}>
    <ListItemText
        editMarkerName={false}
        markerCoords={undefined}
        markerName={undefined}
        markerNumber={undefined}
        setNewMarkerName={[Function]}
        switchEditMarkerName={[Function]} />
    <Buttons
        editMarkerName={false}
        markerIndex={undefined}
        setDeleteMarkerIndex={undefined}
        switchEditMarkerName={[Function]} />
</styled.div>,
  "nodes": Array [
    <styled.div
      current={undefined}
      onClick={[Function]}>
      <ListItemText
            editMarkerName={false}
            markerCoords={undefined}
            markerName={undefined}
            markerNumber={undefined}
            setNewMarkerName={[Function]}
            switchEditMarkerName={[Function]} />
      <Buttons
            editMarkerName={false}
            markerIndex={undefined}
            setDeleteMarkerIndex={undefined}
            switchEditMarkerName={[Function]} />
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <ListItem
        setCurrentMarker={[Function]}
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ListItem {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "setCurrentMarker": [Function],
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": Object {
          "canSendMarkerName": false,
          "editMarkerName": false,
          "newMarkerName": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div
          current={undefined}
          onClick={[Function]}>
          <ListItemText
                    editMarkerName={false}
                    markerCoords={undefined}
                    markerName={undefined}
                    markerNumber={undefined}
                    setNewMarkerName={[Function]}
                    switchEditMarkerName={[Function]} />
          <Buttons
                    editMarkerName={false}
                    markerIndex={undefined}
                    setDeleteMarkerIndex={undefined}
                    switchEditMarkerName={[Function]} />
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div
          current={undefined}
          onClick={[Function]}>
          <ListItemText
                    editMarkerName={false}
                    markerCoords={undefined}
                    markerName={undefined}
                    markerNumber={undefined}
                    setNewMarkerName={[Function]}
                    switchEditMarkerName={[Function]} />
          <Buttons
                    editMarkerName={false}
                    markerIndex={undefined}
                    setDeleteMarkerIndex={undefined}
                    switchEditMarkerName={[Function]} />
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ListItem
    setCurrentMarker={[Function]}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    } />,
}
`;
